### PR TITLE
Update parser.go

### PR DESCRIPTION
**Pull request by moredure**

### CODE REVIEW

File: robotstxt.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func (g *Group) findRule(path string) (ret *rule) {
 				// From Google's spec:
 				// The order of precedence for rules with wildcards is undefined.
 				if l := len(r.pattern.String()); l > prefixLen {
-					prefixLen = len(r.pattern.String())
+					prefixLen = l
```
Comment (temoto) [inline robotstxt.go:211]: this one looks good, thanks

**Comment by temoto** [comment]
replacer/asterisk is very confusing change. If this is a draft work, you can mark pull request as draft with a button here.

**Comment by moredure** [comment]
In practice TrimRight using TrimRightFunc internally, just one memory allocation less, replacer also should allocate less. Also we can use EqualFold to reduce one more memory allocation ToLower, I guess User-Agent/Crawl-Delay tokens very frequently used in camelcase, in parseLine function.

**Comment by moredure** [comment]
```
BenchmarkFromBytes/One_Replacer
BenchmarkFromBytes/One_Replacer-4         	 7473553	       149 ns/op	      48 B/op	       3 allocs/op
BenchmarkFromBytes/Two_strings.Replace
BenchmarkFromBytes/Two_strings.Replace-4  	 8631030	       139 ns/op	      24 B/op	       4 allocs/op
```

**Comment by moredure** [comment]
```go
package robotstxt

import (
	"strings"
	"testing"
)

func BenchmarkReplacer(b *testing.B) {
	t2 := `sdfsd \*\$\* sdfsdf`
	b.Run("One Replacer", func(b *testing.B) {
		b.ReportAllocs()
		for i := 0; i < b.N; i ++  {
			r.Replace(t2)
		}
	})

	b.Run("Two strings.Replace", func(b *testing.B) {
		b.ReportAllocs()
		for i := 0; i < b.N; i ++  {
			strings.Replace(strings.Replace(t2, `\*`, `.*`, -1), `\$`, `$`, -1)
		}
	})
}


func BenchmarkTrimFunc(b *testing.B) {
	b.Run("TrimRightFunc", func(b *testing.B) {
		b.ReportAllocs()
		for i := 0; i < b.N; i ++  {
			strings.TrimRightFunc("sdfsdffsdfsdf*", star)
		}
	})

	b.Run("TrimRight", func(b *testing.B) {
		b.ReportAllocs()
		for i := 0; i < b.N; i ++  {
			strings.TrimRight("sdfsdffsdfsdf*", "*")
		}
	})
}
```

File: robotstxt.go (hunk 1 above)
Comment (moredure) [reply robotstxt.go:211]: Two nanosecond less, pattern.String() just returning backing expr so no allocations, please look to TrimRightFunc, Replacer and asterisk and to strings.EqualFold.

**Comment by moredure** [comment]
Looks like for replacer benchmark is worser when no symbols to replace

**Comment by moredure** [comment]
In worst case scenario (last switch case) equal-fold might be a little bit slower.

**Comment by temoto** [comment]
> Check for yourself:
> 
> ```go
> package robotstxt
> func BenchmarkReplacer(b *testing.B) {
> func BenchmarkTrimFunc(b *testing.B) {
> func BenchmarkEqualFold(b *testing.B) {
> ```

This should be in patch.

**Comment by moredure** [comment]
Let's start from small changes)

**Comment by moredure** [comment]
@temoto

**Pull request merged**